### PR TITLE
Fix invalid system parameter schema

### PR DESCRIPTION
**Pull request by nllong**
#### Any background context you want to provide?

The system parameter schema was malformed. 

#### What does this PR accomplish?
* Fix JSON schema to conform to standard
* Add ETS to the schema
* Fix spawn ETS coupling
* Fix warnings of unclosed files
* Update dependencies (GeoJSON and JSONSchema)

#### How should this be manually tested?
* `py.test` should cover functionality
* Code review

#### What are the relevant tickets?
Address #87

### CODE REVIEW

File: geojson_modelica_translator/geojson/urbanopt_geojson.py [hunk 1]
```diff
@@ -58,7 +58,8 @@ class UrbanOptGeoJson(object):
 
     def __init__(self, filename):
         if os.path.exists(filename):
-            self.data = geojson.load(open(filename))
+            with open(filename, "r") as f:
```
Comment (nllong) [inline geojson_modelica_translator/geojson/urbanopt_geojson.py:61]: fixes the unclosed file warning

File: geojson_modelica_translator/model_connectors/spawnBui_ETS_Coupling.py [hunk 2]
```diff
@@ -183,21 +181,22 @@ def to_modelica(self, scaffold):
...
+                )
+                ets_data = None
+                if ets_model_type == "Indirect Cooling":
+                    ets_data = self.system_parameters.get_param_by_building_id(
+                        building["building_id"],
+                        "ets_model_parameters.indirect_cooling"
+                    )
+                    # this is a complete hack (NL is culprit), do we really need the data in arrays? If so, then we
+                    # would need to update the schema.
+                    for k in ets_data.keys():
```
Comment (nllong) [inline geojson_modelica_translator/model_connectors/spawnBui_ETS_Coupling.py:195]: This is a hack that needs to be resolved in #97. Do not fix now.

File: geojson_modelica_translator/system_parameters/schema.json [hunk 3]
```diff
@@ -242,21 +237,76 @@
...
       "additionalProperties": false
     },
+    "ets_indirect_cooling_parameters": {
+      "description": "Indirect cooling ETS. The energy transfer is implemented through nondirect contact of water loops. Parameters associated with indirect cooling ETS",
+      "type": "object",
+      "properties": {
+        "Q_Flow_Nominal": {
+          "description": "Nominal flow rate (m3/s [verify])",
+          "type": "number",
+          "default": 8000
```
Comment (nllong) [inline geojson_modelica_translator/system_parameters/schema.json:247]: note that the defaults are not respected in the jsonschema python package. They are mostly just examples of how the data look.

File: geojson_modelica_translator/system_parameters/system_parameters.py [hunk 4]
```diff
@@ -41,9 +41,9 @@ class SystemParameters(object):
     """
 
     PATH_ELEMENTS = [
-        {"jsonpath": "$.buildings.*[?load_model=Spawn].load_model_parameters.spawn.idf_filename"},
-        {"jsonpath": "$.buildings.*[?load_model=Spawn].load_model_parameters.spawn.epw_filename"},
-        {"jsonpath": "$.buildings.*[?load_model=Spawn].load_model_parameters.spawn.mos_weather_filename"}
+        {"json_path": "$.buildings.*[?load_model=Spawn].load_model_parameters.spawn.idf_filename"},
```
Comment (nllong) [inline geojson_modelica_translator/system_parameters/system_parameters.py:44]: just cleaning this up.

File: tests/model_connectors/test_ets.py [hunk 5]
```diff
@@ -28,45 +28,64 @@
...
-        self.assertIsNotNone(self.ets)
-
-        return self.ets
+    def setUp(self):
+        # self.data_dir = os.path.join(os.path.dirname(__file__), 'data')
+        # self.output_dir = os.path.join(os.path.dirname(__file__), 'output')
+        # if not os.path.exists(self.output_dir):
+        #     os.makedirs(self.output_dir)
+
+        # base_folder = os.path.join(os.getcwd(), "geojson_modelica_translator")
```
Comment (nllong) [inline tests/model_connectors/test_ets.py:50]: ETSTemplate isn't used at the moment. If we want it to be used, then need to update how this works as this is reading from the schema and not instances.

File: geojson_modelica_translator/system_parameters/system_parameters.py [hunk 6]
```diff
@@ -123,7 +145,10 @@ def get_param(self, path, data=None, default=None):
         else:
             value = data.get(check_path, None)
             if value is None:
-                return default
+                if impute_default:
```
Comment (nllong) [inline geojson_modelica_translator/system_parameters/system_parameters.py:148]: new logic to not determine the default from the data in the buildings.default section.

File: geojson_modelica_translator/geojson/urbanopt_geojson.py (hunk 1 above)
Comment (YanfeiNREL) [reply geojson_modelica_translator/geojson/urbanopt_geojson.py:61]: this is good!

File: tests/model_connectors/test_ets.py (hunk 5 above)
Comment (YanfeiNREL) [reply tests/model_connectors/test_ets.py:50]: self.ets = ETSTemplate(
            self.thermal_junction_properties_geojson,
            self.system_parameters_geojson,
            self.ets_from_building_modelica,
        )

Comment (nllong) [reply tests/model_connectors/test_ets.py:50]: yeah, but ETSTemplate isn't used in the main code (just a unit test). So I've removed it until it is connected to something.

**Comment by nllong** [comment]
> There are 10 tests failed: (1) There are 2 failures related to ets-template, which is related with schema update and i can fix later; (2)The other 8 failures are all related with Docker(Docker not configured on host computer, unable to run). But my laptop is not supporting docker currently. I can skip the Docker.

This should run on windows too. Did you export your MODELICAPATH env var to point to the MBL directory? 

How were you still seeing the other 2 failures? Travis did not see those errors.

**Pull request merged**